api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxMlCohortQueries.java

```diff
@@ -129,4 +155,190 @@ public CohortDefinition getPatientsWhoMissedNextAppointmentAndNotTransferredOut(
+            hivMetadata.getTypeOfVisitConcept().getConceptId(),
+            hivMetadata.getPatientFoundConcept().getConceptId(),
+            hivMetadata.getBuscaConcept().getConceptId(),
+            hivMetadata.getPatientFoundYesConcept().getConceptId()));
+
+    return sqlCohortDefinition;
+  }
+
+  // a and b and Traced (Unable to locate)
+  public CohortDefinition getPatientsWhoMissedNextAppointmentAndNotTransferredOutAndTraced() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+
+    cd.setName(
+        "Get patients who missed next appointment, not transferred out and traced (Unable to locate)");
+    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+
+    cd.addSearch(
+        "missedAppointmentLessTransfers",
+        EptsReportUtils.map(
+            getPatientsWhoMissedNextAppointmentAndNotTransferredOut(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "patientsNotFound",
+        EptsReportUtils.map(
+            getPatientsTracedAndNotFound(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "patientsFound",
+        EptsReportUtils.map(
+            getPatientTracedAndFound(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "scheduledLastAppointment",
+        EptsReportUtils.map(
+            getPatientsLastScheduledAppointmentOrDrugPickup(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+
+    cd.setCompositionString(
+        "missedAppointmentLessTransfers AND ((patientsNotFound AND scheduledLastAppointment) AND NOT (patientsFound AND scheduledLastAppointment))");
+
+    return cd;
+  }
+
+  /*
+   a and b and Untraced Patients
+  */
+  public CohortDefinition getPatientsWhoMissedNextAppointmentAndNotTransferredOutAndUntraced() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+
+    cd.setName("Get patients who missed next appointment, not transferred out and untraced");
+    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+
+    cd.addSearch(
+        "missedAppointmentLessTransfers",
+        EptsReportUtils.map(
+            getPatientsWhoMissedNextAppointmentAndNotTransferredOut(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "withoutVisitCard",
+        EptsReportUtils.map(
+            getAllPatientsWithoutVisitCard(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
```

Comment:
@KelvinWanyama Shouldn't the date boundaries for `All patients without patient visit card` be `the last scheduled appointment or drugs pick date` and `reporting end date`?
I don't see how its limiting by last scheduled appointment.

Reply:
When you look at the composition string, I am ANDing them with the scheduled last appointments or drug pick up.
I wrote separate queries and then joined them through the composition string.